src/core/event/group.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2001-2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef XLIO_GROUP_H
+#define XLIO_GROUP_H
+
+#include <vector>
+
+#include "xlio.h"
+
+/* Forward declarations */
+struct xlio_group_attr;
+class ring;
+class event_handler_manager_local;
+class tcp_timers_collection;
+class sockinfo_tcp;
+
+class xlio_group {
+public:
+    xlio_group(const struct xlio_group_attr *attr);
+    ~xlio_group();
+
+    void poll();
+
+    void add_dirty_socket(sockinfo_tcp *si);
+    void flush();
+
+    void add_ring(ring *);
+    void del_ring(ring *);
+
+    inline unsigned get_flags() { return m_flags; }
+    inline event_handler_manager_local *get_event_handler() { return m_event_handler; }
+    inline tcp_timers_collection *get_tcp_timers() { return m_tcp_timers; }
+
+public:
+    xlio_socket_event_cb_t m_socket_event_cb;
+    xlio_socket_comp_cb_t m_socket_comp_cb;
+    xlio_socket_rx_cb_t m_socket_rx_cb;
+
+private:
+    std::vector<ring *> m_rings;
+    event_handler_manager_local *m_event_handler;
+    tcp_timers_collection *m_tcp_timers;
+
+    std::vector<sockinfo_tcp *> m_dirty_sockets;
+
+    unsigned m_flags;
```

Comment:
I suggest to call it something like m_ring_group_flags. It really makes it easier to do simple text search.

Reply:
Done, renamed to m_group_flags